src/action.py

```diff
@@ -3,30 +3,69 @@
 import click
 import os
 from models.config import Config
+from collections import defaultdict
+from itertools import chain
 
 class ClickGlob:
-    def command(click_group, namespace):
-        def __glob_actions(namespace):
-            parts = ['/var/lib/adminware/tools',
-                     namespace, '*/config.yaml']
-            paths = glob.glob(os.path.join(*parts))
-            return list(map(lambda x: Action(x), paths))
+    def __glob_actions(namespace):
+        parts = ['/var/lib/adminware/tools',
+                 namespace, '*/config.yaml']
+        paths = glob.glob(os.path.join(*parts))
+        return list(map(lambda x: Action(x), paths))
 
+    def command(click_group, namespace):
         # command_func is either run_open or run_batch, what this is decorating
         def __command(command_func):
-            actions = __glob_actions(namespace)
+            actions = ClickGlob.__glob_actions(namespace)
             for action in actions:
                 action.create(click_group, command_func)
 
         return __command
 
+    def command_family(click_group, namespace):
+        def __command_family(command_func):
+            actions = ClickGlob.__glob_actions(namespace)
+            families = []
+            for action in actions:
+                if action.get_families(): families += [action.get_families()]
+            families_dict = __combine_dicts(families)
+            for key in families_dict.keys():
+                __create_option(click_group, command_func, key, families_dict[key])
+
+        # when want to combine the family dictionaries - while concatenating duplicate values
+        def __combine_dicts(dicts):
```

Comment:
I am struggling to follow what is happening in this section of code. The number of `dicts` is confusing, do we need to make so many/ merge them in this way?

Reply:
All that's going on is that a dict is being created for each tool - with a key for each family that tool's in and a value that's simply the tool. These are then merged to create a dict with a key for each family and values for each tool in that family. However I can change this to something more logical to you, more discussion below.